wsagent/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitConnection.java

```diff
@@ -484,4 +484,14 @@
 
     /** Set publisher for git output, e.g. for sending git command output to the client side. */
     void setOutputLineConsumerFactory(LineConsumerFactory outputPublisherFactory);
+
+
+    /**
+     * Get the current branch on the current directory
+     *
+     * @return the name of the branch
+     * @throws GitException
+     *         if any exception occurs
+     */
+    String getBranchName() throws GitException;
```

Comment:
IMHO `getBranchName` looks unclear enough. I think I'd much better to rename it to `getCurrentBranchName` or `getCurrentBranch` WDYT?